Subject: [PATCH] dwc2: log failed dwc2_hsotg_core_connect when binding gadget

When binding a gadget to UDC, it is expected that BSESVLD is true immediately or arrives shortly after within an interrupt, after which dwc2_hsotg_core_connect is retried.
If the PHY is misconfigured, BSESVLD may never arrive, causing confusion without any useful debug messages.

diff --git a/drivers/usb/dwc2/gadget.c b/drivers/usb/dwc2/gadget.c
index 834fc02610a2..2ef3491b1990 100644
--- a/drivers/usb/dwc2/gadget.c
+++ b/drivers/usb/dwc2/gadget.c
@@ -3600,8 +3600,13 @@ void dwc2_hsotg_core_disconnect(struct dwc2_hsotg *hsotg)
 void dwc2_hsotg_core_connect(struct dwc2_hsotg *hsotg)
 {
 	/* remove the soft-disconnect and let's go */
-	if (!hsotg->role_sw || (dwc2_readl(hsotg, GOTGCTL) & GOTGCTL_BSESVLD))
-		dwc2_clear_bit(hsotg, DCTL, DCTL_SFTDISCON);
+	if (hsotg->role_sw && !(dwc2_readl(hsotg, GOTGCTL) & GOTGCTL_BSESVLD)) {
+		dev_info(hsotg->dev, "dwc2_hsotg_core_connect: lack of BSESVLD");
+		return;
+	}
+
+	dev_info(hsotg->dev, "dwc2_hsotg_core_connect: success");
+	dwc2_clear_bit(hsotg, DCTL, DCTL_SFTDISCON);
 }
 
 /**
